src/corelibs/U2View/src/ov_sequence/view_rendering/DetViewMultiLineRenderer.cpp

```diff
@@ -59,13 +59,25 @@ float DetViewMultiLineRenderer::posToXCoordF(const qint64 p, const QSize &canvas
     return commonMetrics.charWidth * (p % symbolsPerLine);
 }
 
-U2Region DetViewMultiLineRenderer::getAnnotationYRange(Annotation *a, int r, const AnnotationSettings *as, const QSize &canvasSize, const U2Region &visibleRange) const {
-    if (qgetenv(ENV_GUI_TEST).toInt() == 1) {
-        U2Region res = singleLinePainter->getAnnotationYRange(a, r, as, QSize(canvasSize.width(), getOneLineHeight()), visibleRange);
-        res.startPos += INDENT_BETWEEN_LINES / 2;
-        return res;
+U2Region DetViewMultiLineRenderer::getAnnotationYRange(Annotation *annotation, int locationRegionIndex, const AnnotationSettings *annotationSettings, const QSize &canvasSize, const U2Region &visibleRange) const {
+    SAFE_POINT(locationRegionIndex < annotation->getRegions().length(), "Invalid locationRegionIndex", U2Region());
+
+    // Compute region offset within a single line.
+    U2Region yRegion = singleLineRenderer->getAnnotationYRange(annotation, locationRegionIndex, annotationSettings, canvasSize, visibleRange);
```

Comment:
Single line canvasSize should be passed to the method. It's height is less.

Reply:
I agree and I plan to remove this method completely in the next CR (see bug description)
Right now I can't change its signature because this is a virtual method used in multiple places.